src/enum_glob_use.rs

```diff
@@ -51,8 +51,9 @@ impl EnumGlobUse {
                             }
                         }
                     } else {
-                        if let Some(dp) = cx.sess().cstore.def_path(def.def_id()).last() {
-                            if let DefPathData::Type(_) = dp.data {
+                        let dp = cx.sess().cstore.relative_def_path(def.def_id());
+                        if  let Some(dpa) = dp.data.last() {
+                            if let  DefPathData::TypeNs(_) = dpa.data {
```

Comment:
Also, I think this whole thing is bogus, why are you not just checking for `Def::Enum` in the first place?